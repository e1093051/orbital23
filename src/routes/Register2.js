import React, { useState } from 'react';
import { useNavigation, useRoute } from '@react-navigation/native';

import {
  StyleSheet,
  Text,
  View,
  Image,
  Button,
  TextInput,
  TouchableOpacity,
  Dimensions
} from 'react-native';

export default () => {
  const navigation = useNavigation();
  const route = useRoute();

  const { name } = route.params;
  const [email, setEmail] = useState("");

  const handleNext = () => {
    const isValidEmail = email.endsWith('@u.nus.edu');
    if (isValidEmail) {
      navigation.navigate('Register3', { name, email });
    } else {
      alert("Please enter a valid NUS email (ends with @u.nus.edu).");
    }
  };
 
  return (
    <View style={styles.bigContainer}>
      <View style={styles.container}>
        <Text style={styles.mainText}>What's your NUS email?</Text>
        <Text style={styles.usual}>
          Enter your NUS webmail (ends with @u.nus.edu) to receive the verification code.
          We will not show this to other users.
        </Text>
        <TextInput
          style={{ height: 40, width: Dimensions.get('window').width - 20, borderWidth: 1, margin: 10 }}
          value={email}
          onChangeText={setEmail}
        />
        <TouchableOpacity
          activeOpacity={0.75}
          style={styles.signUpButton}
          onPress={handleNext}
        >
          <Text style={styles.signUpText}>Next</Text>
        </TouchableOpacity>
      </View>
      <TouchableOpacity
        activeOpacity={0.75}
        style={styles.buttonContainer}
        onPress={() => navigation.navigate('Login')}
      >
        <Text style={styles.registerText}>Log in</Text>
      </TouchableOpacity>
    </View>
  );
}


const styles = StyleSheet.create({
  bigContainer: {
    flex: 1,
    alignItems: 'center',
    backgroundColor: '#FFFFFF',
  },
  container: {
    paddingTop: 10,
    justifyContent: 'flex-start',
    alignItems: 'flex-start',
    backgroundColor: '#FFFFFF',
  },
  mainText: {
    color: 'black',
    fontWeight: 'bold',
    fontSize: 26,
    margin: 10
  },
  usual: {
    color: 'black',
    fontSize: 14,
    margin: 3,
    left: 7
  },
  small: {
    color: 'gray',
    fontSize: 14
  },
  signUpButton: {
    height: 40,
    width: Dimensions.get('window').width - 20,
    backgroundColor: '#2de0ff',
    borderRadius: 5,
    justifyContent: 'center',
    alignItems: 'center',
    left: 10,
    margin: 0
  },
  signUpText: {
    color: 'white',
    fontWeight: 'bold',
    justifyContent: 'center',
    alignItems: 'center',
  },
  buttonContainer:{
    justifyContent: 'center',
    alignItems: 'center',
  },
  buttonContainer: {
    position: 'absolute',
    bottom: 15,
    backgroundColor: 'white',
    borderRadius: 2,
    justifyContent: 'center',
    alignItems: 'center',
    margin: 10
  },
  registerText: {
    color: '#2de0ff',
    fontWeight: 'bold',
  },
});