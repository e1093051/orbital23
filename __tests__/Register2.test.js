import React from 'react';
import renderer from 'react-test-renderer';
import Register2 from '../src/routes/Register2';
import { fireEvent, render, waitFor } from '@testing-library/react-native'; 

// Mock the alert function
global.alert = jest.fn();

jest.mock('@react-navigation/native', () => {
  return {
    useNavigation: () => ({
      navigate: jest.fn(),
      replace: jest.fn(),
    }),
    useRoute: () => ({
      params: { name: 'John' }, // A dummy name for the test
    }),
  };
});

describe('<Register2 />', () => {
  it('navigates to Register3 with a valid email', async () => {
    const { getByTestId, getByText } = render(<Register2 />); // Render the component

    // Test valid email addresses
    const validEmails = [
      'john@u.nus.edu',
      'jane.smith@u.nus.edu',
      'hello123@u.nus.edu',
    ];

    for (const email of validEmails) {
      const textInput = getByTestId('nextButton'); // Get the TextInput element by testID
      fireEvent.changeText(textInput, email); // Set the email value

      const nextButton = getByText('Next'); // Get the Next button element by its text
      fireEvent.press(nextButton); // Press the Next button

      await waitFor(() => {
        // Ensure that the 'navigate' function was called with the correct arguments
        expect(require('@react-navigation/native').useNavigation().navigate).toHaveBeenCalledWith('Register3', {
          name: 'John', // Pass the name along with the email
          email,
        });
      });

      // Check if alert was called
      expect(global.alert).toHaveBeenCalledWith(
        "Please enter a valid NUS email (ends with @u.nus.edu)."
      );
    }
  });
});
