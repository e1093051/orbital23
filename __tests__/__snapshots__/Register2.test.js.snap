// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<View
  style={
    {
      "alignItems": "center",
      "backgroundColor": "#FFFFFF",
      "flex": 1,
    }
  }
>
  <View
    style={
      {
        "alignItems": "flex-start",
        "backgroundColor": "#FFFFFF",
        "justifyContent": "flex-start",
        "paddingTop": 10,
      }
    }
  >
    <Text
      style={
        {
          "color": "black",
          "fontSize": 26,
          "fontWeight": "bold",
          "margin": 10,
        }
      }
    >
      What's your NUS email?
    </Text>
    <Text
      style={
        {
          "color": "black",
          "fontSize": 14,
          "left": 7,
          "margin": 3,
        }
      }
    >
      Enter your NUS email address (ends with @u.nus.edu) to receive the verification code. We will not show this to other users.
    </Text>
    <TextInput
      onChangeText={[Function]}
      style={
        {
          "borderWidth": 1,
          "height": 40,
          "margin": 10,
          "width": 730,
        }
      }
      value=""
    />
    <View
      accessibilityState={
        {
          "busy": undefined,
          "checked": undefined,
          "disabled": undefined,
          "expanded": undefined,
          "selected": undefined,
        }
      }
      accessibilityValue={
        {
          "max": undefined,
          "min": undefined,
          "now": undefined,
          "text": undefined,
        }
      }
      accessible={true}
      collapsable={false}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        {
          "alignItems": "center",
          "backgroundColor": "#2de0ff",
          "borderRadius": 5,
          "height": 40,
          "justifyContent": "center",
          "left": 10,
          "margin": 0,
          "opacity": 1,
          "width": 730,
        }
      }
    >
      <Text
        style={
          {
            "alignItems": "center",
            "color": "white",
            "fontWeight": "bold",
            "justifyContent": "center",
          }
        }
      >
        Next
      </Text>
    </View>
  </View>
  <View
    accessibilityState={
      {
        "busy": undefined,
        "checked": undefined,
        "disabled": undefined,
        "expanded": undefined,
        "selected": undefined,
      }
    }
    accessibilityValue={
      {
        "max": undefined,
        "min": undefined,
        "now": undefined,
        "text": undefined,
      }
    }
    accessible={true}
    collapsable={false}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      {
        "alignItems": "center",
        "backgroundColor": "white",
        "borderRadius": 2,
        "bottom": 15,
        "justifyContent": "center",
        "margin": 10,
        "opacity": 1,
        "position": "absolute",
      }
    }
  >
    <Text
      style={
        {
          "color": "#2de0ff",
          "fontWeight": "bold",
        }
      }
    >
      Log in
    </Text>
  </View>
</View>
`;
